bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/api/ViessmannApi.java

```diff
@@ -288,8 +288,14 @@ private boolean executePost(String url, String json) {
                                 String.format("API Call limit reached. Reset at {}",
                                         viError.getExtendedPayload().getLimitRestetDateTime()));
                     } else {
-                        logger.warn("ViError: {} | Reason: {}", viError.getMessage(),
-                                viError.getExtendedPayload().getReason());
+                        if (viError.getExtendedPayload() != null) {
+                            logger.warn("ViError: {} | Reason: {} {}", viError.getMessage(),
+                                    viError.getExtendedPayload().getReason(),
+                                    viError.getExtendedPayload().getDetails());
+                        } else {
+                            logger.warn("ViError: {}", viError.getMessage());
```

Comment:
Is this the only place where you use these information? You could override `ExtendedPayload.toString` with something like


```
@Override
public String toString() {
    return Objects.requireNonNullElse(reason, "") + " " + Objects.requireNonNullElse(details, "");
```

and simplify the above code to 

```
logger.warn("ViError: {} | Reason: {}", viError.getMessage(), viError.getExtendedPayload());
```

If the extended payload is null, this will print "null", otherwise the same information as before.